### PR TITLE
:seedling: Bump CAPI to v1.8.2

### PR DESCRIPTION
**Pull request by mquhuy**
Bumping CAPI to v1.8.2

### CODE REVIEW

File: test/go.mod [hunk 1]
```diff
@@ -21,7 +21,7 @@ require (
 	k8s.io/client-go v0.30.4
 	k8s.io/klog/v2 v2.120.1
 	k8s.io/utils v0.0.0-20231127182322-b307cd553661
-	sigs.k8s.io/cluster-api v1.8.1
+	sigs.k8s.io/cluster-api v1.8.2
 	sigs.k8s.io/cluster-api/test v1.8.1
```
Comment (adilGhaffarDev) [inline test/go.mod:25]:
```suggestion
	sigs.k8s.io/cluster-api/test v1.8.2
```

**Review by tuminoid — COMMENTED** [review]
/approve

**Pull request merged**